Add module and script to handle blob upload

smarlowucf opened the pull request.

smarlowucf committed:
Add module and script to handle blob upload

rjschwei commented on gceuploadblob:
```diff
@@ -0,0 +1,108 @@
+    required=True
+)
+argparse.add_argument(
+    '--blob-name',
```
Maybe this should be `source-name` ?

smarlowucf replied:
See discussion below

rjschwei commented on gceuploadblob:
```diff
@@ -0,0 +1,108 @@
+    required=True
+)
+argparse.add_argument(
+    '--filename',
```
Maybe `target-filename`?

smarlowucf replied:
See discussion below

rjschwei commented on gceuploadblob:
```diff
@@ -0,0 +1,108 @@
+argparse.add_argument(
+    '--filename',
+    dest='filename',
+    help='The path to the file to be uploaded',
```
The help is not correct, if I read it correctly the intention is that this argument is the filename used when the upload is complete.

There should only be the need for one argument for the source of what is going to be uploaded. If `blob-name` is that source argument then `filename` appears redundant or vice versa.

smarlowucf replied:
I think both arguments are useful. `--blob-name` could be optional though. If blob name is not provided then the filename is used. We could make it more verbose with `--source-filename` or something similar.

smarlowucf replied:
Updated

rjschwei commented on gceuploadblob:
```diff
@@ -0,0 +1,108 @@
+    action='store_true',
+    default=False,
+    dest='verbose',
+    help='Enable on verbose output'
```
s/on//

smarlowucf committed:
Update help text

Rename option to source filename and make blob name optional.
If blob name is not provided use the filename for blob name.

rjschwei commented on gceuploadblob:
```diff
@@ -0,0 +1,107 @@
+argparse.add_argument(
+    '--blob-name',
+    dest='blob_name',
+    help='The name to use for the blob. If not provided the filename is used',
```
s/filename/source filename/

smarlowucf replied:
Fixed

smarlowucf committed:
Fix help comment

s/filename/source filename/
